### PR TITLE
feat: add C implementation for `math/base/special/truncn`

### PR DESCRIPTION
**Pull request by gunjjoshi**
## Description

> What is the purpose of this pull request?

This pull request:

-   adds C implementation for [`math/base/special/truncn`](https://github.com/stdlib-js/stdlib/tree/develop/lib/node_modules/%40stdlib/math/base/special/truncn).

```c
double stdlib_base_truncn( const double x, const int32_t n );
```

## Related Issues

> Does this pull request have any related issues?

This pull request:

-   resolves a part of #649.

## Questions

> Any questions for reviewers of this pull request?

No.

## Other

> Any other information relevant to this pull request? This may include screenshots, references, and/or implementation notes.

Similar to what we did in [`roundb`](https://github.com/stdlib-js/stdlib/pull/2544#issuecomment-2218413328), I have remove some test cases here too, which were not applicable.

## Checklist

> Please ensure the following tasks are completed before submitting this pull request.

-   [x] Read, understood, and followed the [contributing guidelines][contributing].

* * *

@stdlib-js/reviewers

[contributing]: https://github.com/stdlib-js/stdlib/blob/develop/CONTRIBUTING.md

### CODE REVIEW

File: lib/node_modules/@stdlib/math/base/special/truncn/README.md [hunk 1]
```diff
@@ -89,6 +89,92 @@ for ( i = 0; i < 100; i++ ) {
...
+<section class="examples">
+
+### Examples
+
+```c
+#include "stdlib/math/base/special/truncn.h"
+#include <stdio.h>
+
+int main( void ) {
+    const double x[] = { 3.14, -3.14, 0.0, 0.0 / 0.0 };
```
Comment (Planeshifter) [inline lib/node_modules/@stdlib/math/base/special/truncn/README.md:159]: @gunjjoshi I suggest to update the example code values here and in the `example.c` file to actually perform rounding.

**Review by Planeshifter — APPROVED** [review]
One small request, otherwise this looks great!

File: lib/node_modules/@stdlib/math/base/special/truncn/README.md [hunk 2]
```diff
@@ -89,6 +89,92 @@ for ( i = 0; i < 100; i++ ) {
...
+
+    double y;
+    int i;
+    for ( i = 0; i < 4; i++ ) {
+        y = stdlib_base_truncn( x[ i ], -2 );
```
Comment (Planeshifter) [inline lib/node_modules/@stdlib/math/base/special/truncn/README.md]: @gunjjoshi I suggest to update the example code values here and in the `example.c` file to actually perform rounding.

Comment (gunjjoshi) [reply lib/node_modules/@stdlib/math/base/special/truncn/README.md]: Updated the values, @Planeshifter.

Comment (Planeshifter) [reply lib/node_modules/@stdlib/math/base/special/truncn/README.md]: Thank you, @gunjjoshi!

**Pull request merged**